k2/csrc/connect.cu

```diff
@@ -113,10 +115,14 @@ class Connector {
                   fsas_idx01x = fsas_row_splits2_data[fsas_idx01],
                   fsas_idx012 = fsas_idx01x + arcs_idx2,
                   fsas_dest_state_idx01 = dest_states_data[fsas_idx012];
-          // if this arc is a self-loop, just ignore this arc as we won't
-          // processe the dest_state (current state) again
+          // 1. If this arc is a self-loop, just ignore this arc as we won't
+          // processe the dest_state (current state) again.
+          // 2. If the state this arc pointing to is accessible, skip it.
+          // 3. If more than one arc leads to the same state, we select only
```

Comment:
Are there unittests covering this case?

Reply:
This case has nothing to do with the result,  all states would finally be marked as accessible or not, no matter how many times we process the state.
We select only one here to avoid out-of-memory issue,  if we **do not** unique the adding states, the states number will explode and eat up all the memory.
I have no idea how to add unit tests.